Fix empty vlan_interfaces and port_channel_interfaces

titom73 opened the pull request:
Provides fixes to not generate empty variables for both port_channel_interfaces and vlan_interfaces.

## Fix

Issue: #213

## Description

- Add condition to only generate vlan_interfaces on l2leaf when they are running in MLAG since AVD does not generate SVI for tenant vlans
- Add condition to only generate port_channel_interfaces on l3leaf when they are running in MLAG mode. We assume l2leaf are only using PO to l3leafs in MLAG and do not leverage PO on a single l3leaf

titom73 committed:
Fix vlan_interfaces - #213

Add condition to generate vlan_interfaces only if l2leaf is running in
mlag.

With AVD, it is expected l2leaf only has SVI for MLAG configuration

titom73 committed:
Fix port_channel_interfaces - #213

Add condition to only generate port_channel_interfaces if running in
MLAG

We can consider devices are not attached to a single l3leaf using
port-channel.

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2:
```diff
@@ -253,6 +253,7 @@ bfd_multihop:
 {% include 'base/bfd-multihop.j2' %}
 
 {# Port-Channel Interfaces #}
+{% if leaf.mlag == true %}
```
There are could some cases where `leaf.mlag == false` and there are still port-channels defined. For Example a port-channel from a server to a single-homed leaf. We should test for all use cases.

titom73 replied:
Yes. In this case, need to add an additional complex test to define if PO is expected or not.

titom73 replied:
Should be resolved with commit 2c610b9

titom73 committed:
Implement Logic for Port Channel Check #213

Implement a logic at l3leaf template to check wether we need
port-channel section or not.

- If device part of MLAG then port-channel section
- If device is part of l2leaf default parent_l3leafs section, then
port-channel section
- If device is listed in at least a nodes-group of parent_l3leafs in l2leaf
group, then port-channel section

carlbuchmann commented:
Closing this PR, as a suitable method to handle variables that are defined but are empty with the `default` filter.

If you want to use the default value when variables evaluate to false or an empty string you have to set the second parameter to `true` -> example: ` {{ my_variable | default({}, true) }}`

https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#defaulting-undefined-variables
